### PR TITLE
feat: DHIS2-9649 Display dropdown menu in which the user can select TEType 

### PR DESCRIPTION
**Pull request by paschalidi**
@JoakimSM 👋 

this is adding the TEType selector to the new page. 
You will see something like this. It is _only_ rendering the component basically.
image

Next PR will take care of the url manipulation.

### CODE REVIEW

File: src/core_modules/capture-core/components/Pages/New/NewPage.component.js [hunk 1]
```diff
@@ -43,21 +71,40 @@ const NewPagePlain = ({
...
+                        <div className={classes.title} >
+                            New {titleText}
+                        </div>
+                        {
+                            (!selectedScopeType || selectedScopeType === scopeTypes.TRACKED_ENTITY_TYPE) &&
+                                <TrackedEntityTypeSelector
+                                    onSelect={handleRegistrationScopeSelection}
+                                    selectedSearchScopeId={selectedScopeId}
+                                />
+                        }
```
Comment (paschalidi) [inline src/core_modules/capture-core/components/Pages/New/NewPage.component.js:96]: Basically here we are adding the component to the view

File: src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js [hunk 2]
```diff
@@ -27,42 +26,9 @@ const buildSearchOption = (id, name, searchGroups, searchScope, type) => ({
...
-                    },
-                };
-            }, {}),
-    [],
-    );
-
-const useSearchOptions = (trackedEntityTypesWithCorrelatedPrograms): AvailableSearchOptions =>
-    useMemo(() =>
+const useSearchOptions = (): AvailableSearchOptions => {
+    const trackedEntityTypesWithCorrelatedPrograms = useTrackedEntityTypesWithCorrelatedPrograms();
```
Comment (paschalidi) [inline src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js:30]: I guess I realised how flexible hooks are and that I didnt really have to pass it as a argument in many cases. I change that in this PR. Take a look and let me know if you see anything strange

File: src/core_modules/capture-core/components/TrackedEntityTypeSelector/TrackedEntityTypeSelector.component.js [hunk 3]
```diff
@@ -59,17 +59,23 @@ const InfoOutlinedIconWithStyles = withStyles({
...
-export const Index =
-  ({ trackedEntityTypesWithCorrelatedPrograms, classes, onSelect, selectedSearchScopeId }: Props) =>
-      (<>
+export const TrackedEntityTypeSelectorPlain =
+  ({ classes, onSelect, selectedSearchScopeId }: Props) => {
+      const trackedEntityTypesWithCorrelatedPrograms = useTrackedEntityTypesWithCorrelatedPrograms();
+
+      const handleSelectionChange = ({ selected }) => {
+          onSelect(selected, scopeTypes.TRACKED_ENTITY_TYPE);
+      };
```
Comment (paschalidi) [inline src/core_modules/capture-core/components/TrackedEntityTypeSelector/TrackedEntityTypeSelector.component.js:68]: Moved this up here coz later on we would like to dispatch some actions from here

**Review by JoakimSM — APPROVED** [review]
Looks good 👍

**Pull request merged**